portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,7 +58,36 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(comment);
+      int max = getRequestNum(request);  
+      response.sendRedirect("/index.html?limit=" + max);
+  }
+
+  private int getRequestNum(HttpServletRequest request) {
+      String limitString = request.getParameter("limit");
+      int limit;
+      try {
+          limit = Integer.parseInt(limitString);
+      } catch(NumberFormatException e) {
+          return 10;
+      }
+      return limit;
```

Comment:
Nit: I would declare, assign, and return `int limit` inside the `try` block so:

```
try {
  int limit = Integer.parseInt(limitString);
  return limit;
} catch (NumberFormatException e) {
  return 10;
}
```

I think it makes it a bit easier to parse the try-catch block.